### PR TITLE
Move cargo-miri into separate package

### PR DESCRIPTION
**Pull request by RalfJung**
So far, `cargo-miri` depended on the Miri library crate. That was mostly for convenience (to avoid having to deal with workspaces), but also because until recently `cargo-miri` needed access to the ["Miri default flags"](https://github.com/rust-lang/miri/blob/af044209b02f561adb3dad9f5f0deea7378f0ac0/src/lib.rs#L75). But with `MIRI_BE_RUSTC` that is no longer the case, so we can remove this fake dependency. :)

@oli-obk what do you think? Also I expect the rustc bootstrap integration will need updates, hints would be appreciated. :D

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,16 @@
 [package]
-authors = ["Scott Olson <scott@solson.me>"]
-description = "An experimental interpreter for Rust MIR."
+authors = ["Miri Team"]
```
Comment (RalfJung) [inline Cargo.toml:2]: We don't have an email address, do we?

**Comment by RalfJung** [comment]
Let's see if this indeed fixes the Windows rebuild weirdness.
@bors try

**Comment by RalfJung** [comment]
No, it still rebuilds winapi...

**Comment by RalfJung** [comment]
Okay I give up, winapi has too many features.^^

**Comment by RalfJung** [comment]
Oh dang... nested workspaces are not a thing so rustc tools cannot use workspaces :(

**Comment by RalfJung** [comment]
I think in that case the only thing we can do is make it two separate packages, and let the `./miri` script do the dispatch to both of them (instead of letting `cargo --all` do that).

**Comment by RalfJung** [comment]
@oli-obk okay I think I got something that works. The corresponding branch to update rustc bootstrap is [here](https://github.com/RalfJung/rust/tree/miri). However, something is odd: `./x.py --stage 0 build src/tools/miri/cargo-miri` does not work, even though the tool is registered [here](https://github.com/RalfJung/rust/blob/0edf516bc915b8a0225986b0dcc501a804cb17b7/src/bootstrap/tool.rs#L653):
```
thread 'main' panicked at 'Error: no rules matched src/tools/miri/cargo-miri.', src/bootstrap/builder.rs:220:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
failed to run: /home/r/src/rust/rustc.3/build/bootstrap/debug/bootstrap --stage 0 build src/tools/miri/cargo-miri
```

File: Cargo.toml (hunk 1 above)
Comment (oli-obk) [reply Cargo.toml:2]: nope

Comment (RalfJung) [reply Cargo.toml:2]: Okay, I guess this works then.

**Comment by RalfJung** [comment]
@Mark-Simulacrum do you have a good idea for why `./x.py --stage 0 build src/tools/miri/cargo-miri` would not work even though that tool is registered with the appropriate macro?

**Comment by RalfJung** [comment]
@oli-obk I verified that `./x.py dist miri` creates a package with both miri and cargo-miri. So I think that `./x.py build` problem is not a blocker.

**Pull request merged**